File: 6-2.rb
#!/usr/bin/env ruby

answers = 0
File.read('6.input').split("\n\n").each do |group|
  num_members = 0
  group_answers = {}
  group.lines.each do |person|
    num_members += 1
    person.strip.each_char do |c|
      group_answers[c] = group_answers[c].to_i + 1
    end
  end
  answers += group_answers.count {|c, v| v == num_members }
end

print("#{answers}\n")

File: 6-2.py
#!/usr/bin/env python

answers = 0
for group in open('6.input').read().split("\n\n"):
     num_members = 0
     group_answers = {}
     for person in group.strip().split("\n"):
          num_members += 1
          for c in person.strip():
               group_answers[c] = group_answers.get(c, 0) + 1

     for c, v in group_answers.items():
          if v == num_members:
               answers += 1

print(f"{answers}")
